cmd/substreams-sink-noop/main.go

```diff
@@ -81,6 +89,108 @@ func run(cmd *cobra.Command, args []string) error {
 		blockRangeArg = args[3]
 	}
 
+	var err error
+	blockmetaUrl := sflags.MustGetString(cmd, "follow-head-blockmeta-url")
+	substreamsSegmentSize := sflags.MustGetUint64(cmd, "follow-head-substreams-segment")
+	reversibleSegmentSize := sflags.MustGetUint64(cmd, "follow-head-reversible-segment")
+	var blockmetaClient pbbmsrvconnect.BlockClient
+	if blockmetaUrl != "" {
+		blockmetaClient = pbbmsrvconnect.NewBlockClient(http.DefaultClient, blockmetaUrl)
+	}
+
+	signalHandler, isSignaled, _ := cli.SetupSignalHandler(0*time.Second, zlog)
+	sessionCounter := uint64(0)
+	stateStorePath := sflags.MustGetString(cmd, "state-store")
+	var sleepingDuration time.Duration
+	for {
+		if blockmetaClient != nil {
+			for {
+				select {
+				case <-ctx.Done():
+					return nil
+				case <-signalHandler:
+					return nil
+				case <-time.After(sleepingDuration):
+					// continue
+				}
+				sleepingDuration = 5 * time.Second
+
+				blockRangeArg, err = computeBlockRangeFromHead(ctx, blockmetaClient, reversibleSegmentSize, substreamsSegmentSize, blockRangeArg)
+				if err != nil {
+					return fmt.Errorf("computing block range from head: %w", err)
+				}
+
+				startBlockString := strings.Split(blockRangeArg, ":")[0]
+				startBlock, err := strconv.Atoi(startBlockString)
+				if err != nil {
+					return fmt.Errorf("converting start block to integer: %w", err)
+				}
+
+				computedEndBlock := strings.Split(blockRangeArg, ":")[1]
+				endBlock, err := strconv.Atoi(computedEndBlock)
+				if err != nil {
+					return fmt.Errorf("converting start block to integer: %w", err)
+				}
+
+				cursorExisting, extractedBlockNumber, err := readBlockNumFromCursor(stateStorePath)
+				if err != nil {
+					return fmt.Errorf("reading start block from state path: %w", err)
+				}
+
+				if cursorExisting {
+					startBlock = int(extractedBlockNumber)
+				}
+
+				if startBlock < endBlock-1 {
+					break
+				}
+
+				zlog.Info("retrying block range computation", zap.Uint64("session_counter", sessionCounter), zap.Int("start_block_computed", startBlock), zap.Int("end_block_computed", endBlock))
```

Comment:
The "retrying" message is unclear, it looks like something failed, but it didn't. Also, start_block and end_block will always be equal, no need to print them both...
Replace with something like:
`"waiting for head block to reach next threshold", zap.Uint64("target", startBlock+substreamsSegmentSize + reversibleSegmentSize), zap.Uint64("current_head", ...` this would be a lot more useful to the guy trying to figure out what's going on!

Reply:
Also, printed every 5 seconds may be too many. Use a counter and only zlog.Info one when `counter % 6 == 0` (so around 30 seconds)
Another approach is to use zlog.Check() to specify debug level at runtime: always zap.DebugLevel until counter%6==0, then use zap.InfoLevel.